js/classes/entities/foreground/loading_animation.js

```diff
@@ -0,0 +1,69 @@
+class LoadingAnimation {
+    constructor() {
+        const center = new Vector(CANVAS.width / 2, CANVAS.height / 2);
+        
+        this.circles = [
+            { x: center.x - 50, y: center.y, dy: 20},
+            { x: center.x, y: center.y, dy: 20},
+            { x: center.x + 50, y: center.y, dy: 20}
+        ];
+        this.currentCircle = 0;
+        this.intervalId = null;
+    }
+
+    start() { 
+        // setInterval() is the reason GameEngine does not control the update-render loop.
+
+        // setInterval() repeatedly calls a function or executes a code snippet, 
+        // with a fixed time delay between each call.
+        // https://developer.mozilla.org/en-US/docs/Web/API/setInterval
+        this.intervalId = setInterval(() => this.update(), 100);
+    }
+
+    stop() {
+        if (this.intervalId) {
```

Comment:
What does this block of code mean?

Reply:
What is intervalId

Reply:
that was bad naming on my part. It's just a reference to the interval itself.